tests/src/performance/Scenario/JitBench/JitBenchHarness.cs

```diff
@@ -0,0 +1,487 @@
+            Console.WriteLine("**********************************************************************");
+        }
+
+        private static void PrintRunningStepInformation(string message)
+        {
+            Console.WriteLine($"-- {message}");
+        }
+
+        private static readonly ScenarioConfiguration s_ScenarioConfiguration;
+
+        private static int s_iteration;
+        private static double[] s_startupTimes;
+        private static double[] s_requestTimes;
+        private static string s_temporaryDirectory;
+        private static string s_jitBenchDevDirectory;
+        private static string s_targetArchitecture;
+
+        /// <summary>
+        /// Provides an interface to parse the command line arguments passed to the JitBench harness.
+        /// </summary>
+        private sealed class JitBenchHarnessOptions
+        {
+            public JitBenchHarnessOptions()
+            {
+                _tempDirectory = Directory.GetCurrentDirectory();
+            }
+
+            [Option('o', Required = false, HelpText = "Specifies the intermediate output directory name.")]
+            public string IntermediateOutputDirectory
+            {
+                get { return _tempDirectory; }
+
+                set
+                {
+                    if (string.IsNullOrWhiteSpace(value))
+                        throw new InvalidOperationException("The intermediate output directory name cannot be null, empty or white space.");
+
+                    if (value.Any(c => Path.GetInvalidPathChars().Contains(c)))
+                        throw new InvalidOperationException("Specified intermediate output directory name contains invalid path characters.");
+
+                    _tempDirectory = Path.IsPathRooted(value) ? value : Path.GetFullPath(value);
+                    Directory.CreateDirectory(_tempDirectory);
+                }
+            }
+
+            [Option("target-architecture", Required = true, HelpText = "JitBench target architecture (It must match the built product that was copied into sandbox).")]
+            public string TargetArchitecture { get; set; }
+
+            public static JitBenchHarnessOptions Parse(string[] args)
+            {
+                using (var parser = new Parser((settings) => {
+                    settings.CaseInsensitiveEnumValues = true;
+                    settings.CaseSensitive = false;
+                    settings.HelpWriter = new StringWriter();
+                    settings.IgnoreUnknownArguments = true;
+                }))
+                {
+                    JitBenchHarnessOptions options = null;
+                    parser.ParseArguments<JitBenchHarnessOptions>(args)
+                        .WithParsed(parsed => options = parsed)
+                        .WithNotParsed(errors => {
+                            foreach (Error error in errors)
+                            {
+                                switch (error.Tag)
```

Comment:
This was brought from the original PR, and it might have been a port from a different project.

---
In reply to: [129456611](https://github.com/dotnet/coreclr/pull/11400#discussion_r129456611) [](ancestors = 129456611)